### DISCUSSION TITLE
Need to allow Zepp OS app to run through browser

### DISCUSSION THREAD

**Discussion by NVCDevelopmentTeam** [zepp-health/zeppos-forum#212] [Developer Feedback]
Hi, I am developing a screen reader application for Zepp OS but because I am blind and facing a lot of financial difficulties as well as accessing and using ZEPP OS devices and emulators like Amatfit, it is very difficult to test the application through an actual device or a simulator. I need your code contribution to develop the project as this is open source so I don't take offense if you make a good contribution to the community. In addition, to facilitate the application experience during application development, I think it is advisable to run the Zepp application through the browser so that we have an overview before proceeding to run on the emulator or device. reality. From there, contribute to making it work stably when publishing the application. I hope, if you are interested, you can contribute here: https://github.com/NVCDevelopmentTeam/Zepp-OS-ScreenReader I look forward to receiving your comments and discussions. Thank.

**Comment by NVCDevelopmentTeam** [comment]
You can read the discussion above.

**Reply by juanitadion** [reply]
@NVCDevelopmentTeam I would like to personally invite you and the team joining the Zepp OS Slack Team room and we should chat more.  I want to help.  https://t.zepp.com/t/zepp-community-slack

**Comment by htoooth** [comment]
Hi, @NVCDevelopmentTeam：
ZeppOS mini programs can currently only run on Zepp OS. This is because, apart from the UI, there are many sensor functionalities that rely on hardware devices, which are not available in browsers. 

And We also do not have browser functionality in Zepp OS. 

You can visit the [Zepp OS Developer Center](https://docs.zepp.com/docs/reference/app-json/) to learn more about the capabilities offered by Zepp OS. In the Developer Center, you will find documentation, API references, sample code, and other development resources related to Zepp OS. These resources will help you understand how to leverage the functionalities of Zepp OS to develop applications.

**Comment by NVCDevelopmentTeam** [comment]
If so, we can't try to build our app when it's not accessible. This clearly shows that ZEPP OS is not a11y compliant and as usual, it should be thanks to the community's help to develop ZEPP-OS-ScreenReader. Also, I think it should be discussed and integrated into smart watch soon. If Zepp OS doesn't allow to run the app on the browser like react or vue or react native then it means that the ability to reach out to know the app or test the app to see if it crashes or not is impossible.

**Comment by NVCDevelopmentTeam** [comment]
I find myself, as a visually impaired user, Zepp OS itself lacks accessibility services and we have to do something about it now.

**Comment by NVCDevelopmentTeam** [comment]
I see wearables running wear os or watch os have their own screen readers like talkback and voiceover. If zepp os has a screen reader it will be better. You can include my repository as a basis to develop it later.
